ingredient/ingredient-kubernetes/src/plugin.ts

```diff
@@ -0,0 +1,31 @@
+import { BaseIngredient, IngredientManager } from "@azbake/core"
+import { promisify } from 'util';
+import { exec as exec_from_child_process } from 'child_process';
+const exec = promisify(exec_from_child_process);
+
+export class KubernetesPlugin extends BaseIngredient {
+
+    public async Execute(): Promise<void> {
+        try {
+            let util = IngredientManager.getIngredientFunction("coreutils", this._ctx)
+            let k8sYamlPath = await this._ingredient.properties.source.valueAsync(this._ctx);
+            let testDeployment = await this._ingredient.properties.parameters.get("testDeployment");
+            let k8sConfigPath = await this._ingredient.properties.parameters.get("config");

```

Comment:
treat all your params as BakeVariables (see your yamlPath above for resolving it)